### ISSUE TITLE
FN RSPEC-3531 : yield embedded in nested non-generator function must not count

### ISSUE THREAD

**Issue by fmallet** [SonarSource/SonarJS#676]
[RSPEC-3531](https://jira.sonarsource.com/browse/RSPEC-3531)
```
function * generatorAndFunction() {                 // FN, Noncompliant

    array.forEach(
        x => { doSomething(); yield; }
    );

}
```

**Comment by fmallet** [comment]
Comes from https://jira.sonarsource.com/browse/SONARJS-866

**Comment by yassin-kammoun-sonarsource** [comment]
Fixed during migration with #2099
